pallets/runtime/tests/src/statistics_test.rs

```diff
@@ -69,13 +118,244 @@ fn investor_count_per_asset_with_ext() {
 
     // Alice sends some tokens to Bob. Token has only one investor.
     unsafe_transfer(alice_did, bob_did, 500);
-    assert_eq!(Statistic::investor_count_per_asset(&ticker), 1);
+    assert_eq!(Statistic::investor_count(&ticker), 2);
 
     // Alice sends some tokens to Charlie. Token has now two investors.
     unsafe_transfer(alice_did, charlie_did, 5000);
-    assert_eq!(Statistic::investor_count_per_asset(&ticker), 2);
+    assert_eq!(Statistic::investor_count(&ticker), 3);
 
     // Bob sends all his tokens to Charlie, so now we have one investor again.
     unsafe_transfer(bob_did, charlie_did, 500);
-    assert_eq!(Statistic::investor_count_per_asset(&ticker), 1);
+    assert_eq!(Statistic::investor_count(&ticker), 2);
+}
+
+#[test]
+fn should_add_tm() {
+    ExtBuilder::default().build().execute_with(|| {
+        let (token_owner_signed, _token_owner_did) =
+            make_account(AccountKeyring::Alice.public()).unwrap();
+
+        let token_name = b"ACME";
+        let ticker = Ticker::try_from(&token_name[..]).unwrap();
+        create_token(token_name, ticker, AccountKeyring::Alice.public());
+
+        let mut tms = Vec::new();
+
+        for i in 0..3u64 {
+            tms.push(TransferManager::CountTransferManager(i));
+        }
+
+        assert_ok!(Statistic::add_transfer_manager(
+            token_owner_signed.clone(),
+            ticker,
+            tms[0].clone()
+        ));
+        assert_eq!(Statistic::transfer_managers(ticker), [tms[0].clone()]);
+
+        assert_noop!(
+            Statistic::add_transfer_manager(token_owner_signed.clone(), ticker, tms[0].clone()),
+            Error::DuplicateTransferManager
+        );
+
+        for i in 1..3u64 {
+            assert_ok!(Statistic::add_transfer_manager(
+                token_owner_signed.clone(),
+                ticker,
+                tms[i as usize].clone()
+            ));
+        }
+
+        assert_eq!(Statistic::transfer_managers(ticker), tms);
+
+        assert_noop!(
+            Statistic::add_transfer_manager(
+                token_owner_signed.clone(),
+                ticker,
```

Comment:
This prefix is common (3x), let's extract to closure taking last bit as a param.